service/src/main/kotlin/io/provenance/digitalcurrency/consortium/stream/EventStreamResponseObserver.kt

```diff
@@ -8,11 +8,7 @@ class EventStreamResponseObserver<T>(private val onNextHandler: (T) -> Unit) : S
     var error: Throwable? = null
 
     override fun onNext(value: T) {
-        try {
-            onNextHandler(value)
-        } catch (t: Throwable) {
-            this.onError(t)
-        }
+        onNextHandler(value)
```

Comment:
This catch was preventing the stream from gracefully shutting down like expected.

Reply:
nice find

Reply:
Did you test this? I'm not sure if these Observer is being used like it was originally intended or not, but originally this was done in a separate thread so calling onError is what sets the error. The awaiting thread would be the one that would have to rethrow the error when it is set, I believe.

Reply:
briefly tested, going to do more tomorrow on a fresh brain. But it now goes into this function:

```
    private fun handleError(t: Throwable) {
        log.error("EventStream error ${t.message}")
        observer.onError(t)
        shutdown()
    }
```

Reply:
This function:

```
private fun startEventLoop() {
        // subscribe to block events tm.event='NewBlock'
        log.info("Opening EventStream websocket")
        lifecycle.onNext(Lifecycle.State.Started)

        subscription = eventStreamService.observeWebSocketEvent()
            .filter {
                if (it is WebSocket.Event.OnConnectionFailed) {
                    handleError(it.throwable)
                }

                it is WebSocket.Event.OnConnectionOpened<*>
            }
            .switchMap {
                log.info("Initializing subscription for tm.event='NewBlock'")
                eventStreamService.subscribe(Subscribe("tm.event='NewBlock'"))
                eventStreamService.streamEvents()
            }
            .filter { !it.result!!.query.isNullOrBlank() && it.result.data.value.block.header.height >= startHeight }
            .map { event -> event.result!! }
            .subscribe(
                { handleEvent(it) },
                { handleError(it) }
            )
    }
```

has the `handleError` in that last subscription line and as far as I could tell in debugging it was never being used because of the try/catch in this observer. The only time that `handleError` got called was if the event stream hadn't seen activity in over 30s.

Reply:
This is behaving as I expect. Thoughts on removing the marker transfer conflict resolution? I'd prefer to take it out.

Reply:
i'd remove it especially since we're also saying the same tx hash could have multiple marker transfers right?